### DISCUSSION TITLE
Need Help in running setup.sh script

### DISCUSSION THREAD

**Discussion by Joaquin144** [oppia/oppia-android#4465] [Q&A (Other)] [answered]
I am just starting out contributing to oppia-android project.
I went through _**Contributing to Oppia Android**_ guide and as told here in https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android#install-oppia-android to run setup.sh file through terminal.
I am getting this error even though every dependency file of setup.sh seems in place.
![image](https://user-images.githubusercontent.com/80385154/180855668-771d738e-8906-4b1e-86f4-9e077943b76f.png)

**Comment by BenHenning** [comment, marked as answer]
Hi @Joaquin144.

I believe ``setup.sh`` needs to be run from the root repository directory, not from within the scripts folder (generally, all build systems and automation are meant to be run at the repository root by default, though there are exceptions where things *can* work in subdirectories).

Therefore, you should try something like:

```sh
cd ~/opensource/oppia-android
sh scripts/setup.sh
```

(Assuming you cloned Oppia Android at ~/opensource/oppia-android).

**Reply by Joaquin144** [reply]
Thanks. It worked.

**Comment by rishujam** [comment]
![Screenshot 2022-11-27 015650](https://user-images.githubusercontent.com/74773876/204107870-b37bfc6d-ab7b-4399-92f7-46a128b79f9d.png)
Is it fine to get this message?

**Reply by rishujam** [reply]
@seanlip

**Reply by BenHenning** [reply]
This is a known issue @rishujam (that the protobuf lint check doesn't work on Windows). For now, you'll need to rely on CI checks for protobuf checks.

**Reply by rishujam** [reply]
Okay thanks
